RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc

```diff
@@ -197,7 +197,7 @@ math::XYZPoint HGCalImagingAlgo::calculatePosition(std::vector<KDNode> &v){
 			 z/total_weight );
     }
   }
-  else {
+  else if (v_size > 0) {
```

Comment:
as an index, it's unsigned, so -1 doesn't work. Given the logical flow of the code, the value will always be set as long as `v_size > 0` and `haloOnlyCluster==true`.